pydantic/main.py

```diff
@@ -646,6 +664,51 @@ def __eq__(self, other: Any) -> bool:
         else:
             return self.model_dump() == other
 
+    def model_copy(self, *, update: dict[str, Any] | None = None, deep: bool = False) -> typing_extensions.Self:
+        """
+        Returns a copy of the model.
+
+        :param update: values to change/add in the new model. Note: the data is not validated before creating
+            the new model: you should trust this data
+        :param deep: set to `True` to make a deep copy of the model
+        :return: new model instance
+        """
+        copied = self.__deepcopy__() if deep else self.__copy__()
+        if update:
+            copied.__dict__.update(update)
+            copied.__fields_set__.update(update.keys())
+        return copied
```

Comment:
> If it's really as simple as adding the update and calling .update(update), I think we may as well as it here.
>
> But if there are complex interactions with include/exclude, I'd prefer a new method.

@samuelcolvin is this simple enough for you? (I think it meets the bar.) It passes all existing tests of copy that don't use include/exclude. (Including checks that you can use this to obtain frozen model instances with modified values.)

I think we should eventually try to get mypy to typecheck the `model_copy` method, but that can come later.

Reply:
@JoshYuJump `model_` is a protected namespace Pydantic V2 uses for its needs. This allows you to have a model that have an attribute named `copy` and avoid clashes with Pydantic public methods.